### ISSUE TITLE
EndPoint routing breaks with ConsumesAttribute on API base controller 

### ISSUE THREAD

**Issue by josundt** [aspnet/AspNetCore#12300]
.NET Core 2.2.301 runtime

In our Web API, we have an abstract base controller class, and all API Controllers inherit from this. We use ApiExplorer for OpenApi support; and in the base controller we used the `ProducesAttribute` and `ConsumesAttribute` attributes to inform ApiExplorer about the _default_ Request and Response body content-types, thereby avoiding the need to use these attributes on every single controller or action method where defaults apply.

```csharp
[ApiController]
[ApiConventionType(typeof(DefaultApiConventions))] // Added after upgrading to 2.2
[Produces("application/json")]
[Consumes("application/json")]
protected abstract class ApiControllerBase : ControllerBase {
    //...
}
```

After upgrading to ASP.NET Core 2.2.301, we noticed that many of the API routes were not hit, just giving a 404. When switching off EndpointRouting it seemed to work again.

**After troubleshooting for a long time I found that when removing the `ConsumesAttribute`  from our base controller, everything started working again.**

PS! I know that the action methods using HTTP Get do not really "consume" anything, but still this techique worked in 2.1, and did not cause any problems for the HTTP Get action methods. It seems to cause problems setting this "globally" when using the new EndpointRouting.

I just wanted to inform about this change in behavior that may be a breaking change with ASP.NET Core 2.2 EndpointRouting for other people using the technique as we did.

Currently, unless you use a `ConsumesAttribute` on every action method that expects a request body paylod, the OpenAPI/Swagger documentation will list **all content types for all InputFormatters as request body format.** 

I think ApiExplorer should offer to set the default content-type for request and response bodies in startup configuration. Consider this a feature request - let me know if you want me to create a new feature request issue for this.

**Comment by mkArtakMSFT** [comment]
Thanks for contacting us, @josundt.
Endpoint Routing differs from the "old" routing system in many aspects. If you really need the "working" behavior, simply disable Endpoint Routing.
@rynowak is this the expected behavior?

@josundt, as for the second issue:
> Currently, unless you use a ConsumesAttribute on every action method that expects a request body paylod, the OpenAPI/Swagger documentation will list all content types for all InputFormatters as request body format.

please file it separately, and we'll review it separately.

**Comment by rynowak** [comment]
The issue with `[Consumes(...)]` and actions without a body has been fixed for 3.0.
